datadog/provider_test.go

```diff
@@ -153,11 +168,25 @@ func testProviderConfigure(r *recorder.Recorder) schema.ConfigureFunc {
 			}
 		}
 		datadogClient := datadog.NewAPIClient(config)
+		//Datadog V2 API config.HTTPClient
+		configV2 := datadogV2.NewConfiguration()
+		configV2.Debug = true
+		configV2.HTTPClient = c
+		if apiURL := d.Get("api_url").(string); apiURL != "" {
+			if strings.Contains(apiURL, "datadoghq.eu") {
+				auth = context.WithValue(auth, datadogV2.ContextServerVariables, map[string]string{
+					"site": "datadoghq.eu",
+				})
+			}
+		}
```

Comment:
I think we need to be smarter here, in case someone is not sending to a datadog URL directly

Reply:
Interesting. How would one configure other `site` for the generated go client? From what I can tell, it seems the possible configurations for the `site` variable on the clients are `datadoghq.com` or `datadoghq.eu`

Reply:
There is another server possible, by setting `ServerIndex` to 2, you can specify an arbitrary URL
See https://github.com/DataDog/datadog-api-spec/blob/master/spec/v1/header.yaml#L22-L29

Reply:
Thanks for that link very helpful! Pushed a commit to better handle non-datadog urls. Let me know what you think.